contracts/near/eth-prover/src/lib.rs

```diff
@@ -136,6 +136,94 @@ impl EthProver {
+        self.check_not_paused(PAUSE_VERIFY);
+        let header: BlockHeader = rlp::decode(header_data.as_slice()).unwrap();
+        let data = Self::verify_trie_proof(header.state_root, key, proof);
+
+        let verification_result = data == account_data;
+        if verification_result && skip_bridge_call {
+            return PromiseOrValue::Value(true);
+        } else if !verification_result {
+            return PromiseOrValue::Value(false);
+        }
+        // Verify block header was in the bridge
+        eth_client::ext(self.bridge_smart_contract.parse().unwrap())
+            .with_static_gas(BLOCK_HASH_SAFE_GAS)
+            .block_hash_safe(header.number)
+            .then(
+                remote_self::ext(env::current_account_id())
+                    .with_static_gas(ON_BLOCK_HASH_GAS)
+                    .on_block_hash(header.hash.unwrap()),
+            )
+            .into()
+    }
+
+    #[result_serializer(borsh)]
+    pub fn verify_storage_proof(
+        &self,
+        #[serializer(borsh)] header_data: Vec<u8>,
+        #[serializer(borsh)] account_proof: Vec<Vec<u8>>, // account proof
+        #[serializer(borsh)] contract_address: Vec<u8>,   // eth address
+        #[serializer(borsh)] account_state: Vec<u8>,      // encoded account state
+        #[serializer(borsh)] storage_key: Vec<u8>,        // keccak256 of storage key
+        #[serializer(borsh)] storage_proof: Vec<Vec<u8>>, // storage proof
+        #[serializer(borsh)] expected_storage_value: Vec<u8>, // storage value
+        #[serializer(borsh)] min_header_height: Option<u64>,
+        #[serializer(borsh)] max_header_height: Option<u64>,
+        #[serializer(borsh)] skip_bridge_call: bool,
+    ) -> PromiseOrValue<bool> {
+        self.check_not_paused(PAUSE_VERIFY);
+        let header: BlockHeader = rlp::decode(header_data.as_slice()).unwrap();
+
+        
+        if let Some(min_header_height) = min_header_height {
+            if header.number < min_header_height {
+                return PromiseOrValue::Value(false);
+            }
+        }
+        
+        if let Some(max_header_height) = max_header_height {
+            if header.number > max_header_height {
+                return PromiseOrValue::Value(false);
+            }
+        }
+        
+        let account_key = near_keccak256(&contract_address).to_vec();
+        let data = Self::verify_trie_proof(header.state_root, account_key, account_proof);
+        
+        let retrieved_storage_hash: Vec<u8> = Rlp::new(&account_state).val_at(2).unwrap();
+        let verification_result = data == account_state;
+        if verification_result {
+            let st_data =
+                Self::verify_trie_proof(H256::from(retrieved_storage_hash), storage_key, storage_proof);
+            let verification_result = st_data == expected_storage_value;
+            if verification_result && skip_bridge_call {
+                return PromiseOrValue::Value(true);
+            }
```

Comment:
`} else if !verification_result { return PromiseOrValue::Value(false) }` ?

Reply:
Fixed in b3fa1f5621c6430a6a12036109038148cba5136f